internal/testutil/k8s.go

```diff
@@ -0,0 +1,64 @@
+package testutil
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+const servicesFile = "services.yml"
+
+func FakeK8sClient() (*fake.Clientset, error) {
+	fakeCs := fake.NewSimpleClientset()
+
+	k8sServicesYaml, err := testDataDir.ReadFile(filepath.Join(testDataRootDir, servicesFile))
+	if err != nil {
+		return nil, fmt.Errorf("reading testdata services.yml: %w", err)
+	}
+
+	var services []*corev1.Service
+	err = yaml.Unmarshal(k8sServicesYaml, &services)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling testata/services.yml: %w", err)
+	}
+
+	// KSM scraper relies on a list of services to add extra data to KSM metrics
+	for _, svc := range services {
+		err := k8sCreateNamespaceIfMissing(fakeCs, svc.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("creating ns %q for service %q: %w", svc.Namespace, svc.Name, err)
+		}
+
+		_, err = fakeCs.CoreV1().Services(svc.Namespace).Create(context.Background(), svc, metav1.CreateOptions{})
```

Comment:
Instead of creating the objects one by one you can pass them as arg to the NewSimpleClientset function:

```
	client := testclient.NewSimpleClientset(getNamespaces()...,getServices()...)
```

It would simplify a lot this function

Reply:
If possible I would remove completely  FakeK8sClient and just call from the differente tests 	
client := testclient.NewSimpleClientset(getNamespaces()...,getServices()...)

Reply:
Interesting solution, I will give a try to creating the services like this.

Reply:
Sovled!